src/main/rx/nrf24_v202.c

```diff
@@ -0,0 +1,295 @@
+    }
+    NRF24L01_SetPower(3);
+    NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);     // Clear data ready, data sent, and retransmit
+    NRF24L01_WriteReg(NRF24L01_11_RX_PW_P0, V2X2_PAYLOAD_SIZE);  // bytes of data payload for pipe 0
+    NRF24L01_WriteReg(NRF24L01_17_FIFO_STATUS, 0x00); // Just in case, no real bits to write here
+    uint8_t rx_tx_addr[] = {0x66, 0x88, 0x68, 0x68, 0x68};
+    NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rx_tx_addr, 5);
+    NRF24L01_WriteRegisterMulti(NRF24L01_10_TX_ADDR, rx_tx_addr, 5);
+
+    delay(50);
+
+    NRF24L01_FlushTx();
+    NRF24L01_FlushRx();
+
+    rf_ch_num = 0;
+
+    // Turn radio power on
+    config |= BV(NRF24L01_00_CONFIG_PWR_UP);
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, config);
+    NRF24L01_SetTxRxMode(RX_EN);
+
+    bind_phase = PHASE_NOT_BOUND;
+    prepare_to_bind();
+
+    switch_channel();
+}
+
+static void set_bound()
+{
+    for (int i = 0; i < V2X2_NFREQCHANNELS; ++i) {
+        rf_channels[i] = freqhopping[i];
+    }
+    rx_timeout = 1000L; // find the channel as fast as possible
+}
+
+static void decode_bind_packet(uint8_t *packet)
+{
+    if ((packet[14] & V2X2_FLAG_BIND) == V2X2_FLAG_BIND) {
+        // Fill out usersettings with bound protocol parameters
+        v2x2_set_tx_id(&packet[7]);
+        // Read usersettings into current values
+        bind_phase = PHASE_BOUND;
+        set_bound();
+    }
+}
+
+// Returns whether the data was successfully decoded
+static nrf24_received_t decode_packet(uint8_t *packet)
+{
+#ifdef DEBUG_V202
+    packet[0] = 230; // throttle = 1900
+    packet[1] = 101; // yaw = 1100
+    packet[2] = 140; // pitch = 1550
+    packet[3] = 25;  // roll = 1400
+#else
+    if(bind_phase != PHASE_BOUND) {
+        decode_bind_packet(packet);
+        return NRF24_RECEIVED_BIND;
+    }
+    // Decode packet
+    if ((packet[14] & V2X2_FLAG_BIND) == V2X2_FLAG_BIND) {
+        return NRF24_RECEIVED_BIND;
+    }
+    if (packet[7] != txid[0] ||
+        packet[8] != txid[1] ||
+        packet[9] != txid[2]) {
+        return NRF24_RECEIVED_NONE;
+    }
+#endif
+    // Restore regular interval
+    rx_timeout = 10000L; // 4ms interval, duplicate packets, (8ms unique) + 25%
+    // TREA order in packet to MultiWii order is handled by
+    // correct assignment to channelindex
+    // Throttle 0..255 to 1000..2000
+    nrf24RcData[v2x2_channelindex[0]] = ((uint16_t)packet[0]) * 1000 / 255 + 1000;
+    for (int i = 1; i < 4; ++i) {
+        uint8_t a = packet[i];
+        nrf24RcData[v2x2_channelindex[i]] = ((uint16_t)(a < 0x80 ? 0x7f - a : a)) * 1000 / 255 + 1000;
+    }
+    uint8_t flags[] = {V2X2_FLAG_LED, V2X2_FLAG_FLIP, V2X2_FLAG_CAMERA, V2X2_FLAG_VIDEO}; // two more unknown bits
+    for (int i = 4; i < 8; ++i) {
+        nrf24RcData[v2x2_channelindex[i]] = 1000 + ((packet[14] & flags[i-4]) ? 1000 : 0);
+    }
+    packet_timer = micros();
+    if (++valid_packets > 50) {
+        bind_phase = PHASE_BOUND;
```

Comment:
this will be be triggered only when `bind_phase == PHASE_BOUND;` - this code does nothing ..